From 5d5dcdc0b1eb421d9eea9e363d862d6e6e64b435 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Fri, 22 Dec 2017 18:04:58 +0800
Subject: [PATCH 3274/5242] MLK-17289-1: hdp: change struct mem variable type
 to pointer

commit  baa7b55845e27b76f03a0970aac9b4322c6e77d2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Change struct mem variable type to pointer in struct state.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hdp/util.c |    8 ++++----
 drivers/mxc/hdp/util.h |    6 +++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/mxc/hdp/util.c b/drivers/mxc/hdp/util.c
index 9b2fdd8..a0c9771 100644
--- a/drivers/mxc/hdp/util.c
+++ b/drivers/mxc/hdp/util.c
@@ -50,28 +50,28 @@
 
 int cdn_apb_read(state_struct *state, u32 addr, u32 *value)
 {
-	struct hdp_mem *mem = &state->mem;
+	struct hdp_mem *mem = state->mem;
 	state->rw->read_reg(mem, addr, value);
 	return 0;
 }
 
 int cdn_apb_write(state_struct *state, u32 addr, u32 value)
 {
-	struct hdp_mem *mem = &state->mem;
+	struct hdp_mem *mem = state->mem;
 	state->rw->write_reg(mem, addr, value);
 	return 0;
 }
 
 int cdn_sapb_read(state_struct *state, u32 addr, u32 *value)
 {
-	struct hdp_mem *mem = &state->mem;
+	struct hdp_mem *mem = state->mem;
 	state->rw->sread_reg(mem, addr, value);
 	return 0;
 }
 
 int cdn_sapb_write(state_struct *state, u32 addr, u32 value)
 {
-	struct hdp_mem *mem = &state->mem;
+	struct hdp_mem *mem = state->mem;
 	state->rw->swrite_reg(mem, addr, value);
 	return 0;
 }
diff --git a/drivers/mxc/hdp/util.h b/drivers/mxc/hdp/util.h
index 6907fe61..c7329b2 100644
--- a/drivers/mxc/hdp/util.h
+++ b/drivers/mxc/hdp/util.h
@@ -132,8 +132,8 @@
 } INTERNAL_MBOX_STATUS;
 
 struct hdp_mem {
-	void __iomem *regs_base;
-	void __iomem *ss_base;
+	void __iomem *regs_base; /* Controller regs base */
+	void __iomem *ss_base; /* HDP Subsystem regs base */
 };
 
 struct hdp_rw_func {
@@ -155,7 +155,7 @@ struct hdp_rw_func {
 	u32 tmp;
 
 	struct mutex mutex;	//mutex may replace running
-	struct hdp_mem mem;
+	struct hdp_mem *mem;
 	struct hdp_rw_func *rw;
 } state_struct;
 /**
-- 
1.7.9.5

